### PR TITLE
Mmt 3551: As a user, I would like to select a provider from my list of provider during draft creation

### PR DESCRIPTION
**Pull request by tle1989**
Create a dropdown, where the user can select a provider from a provider list
 
Screenshot 2024-03-12 at 2 57 55 PM

Screenshot 2024-03-12 at 2 58 08 PM

### CODE REVIEW

File: static/src/js/App.jsx [hunk 1]
```diff
@@ -48,8 +54,54 @@ export const App = () => {
...
+    user, setProviderId, setProviderIds
+  } = useAppContext()
+
+  const [getProviders] = useLazyQuery(GET_ACLS, {
+    variables: {
+      params: {
+        includeFullAcl: true,
+        pageNum: 1,
+        pageSize: 2000,
+        permittedUser: 'typical',
```
Comment (dmistry1) [inline static/src/js/App.jsx:69]: This needs the users actual id

File: static/src/js/App.jsx [hunk 2]
```diff
@@ -48,8 +54,54 @@ export const App = () => {
...
+      const { items } = acls
+
+      if (items.length > 0) {
+        const providerList = items.map(({ acl }) => acl.provider_identity.provider_id)
+
+        setProviderIds(providerList)
+
+        // Check if user does not have providerId
+        // and set it to the first providerId if available
+        if (!user.providerId && providerList.length > 0) {
```
Comment (cgokey) [inline static/src/js/App.jsx:84]: destructure

File: static/src/js/providers/AuthContextProvider/AuthContextProvider.jsx [hunk 3]
```diff
@@ -32,12 +32,16 @@ const AuthContextProvider = ({ children }) => {
   const [cookies, setCookie, removeCookie] = useCookies(['loginInfo', 'launchpadToken', 'data'])
   const { loginInfo = {}, launchpadToken } = cookies
 
-  const setUser = (info) => {
-    setCookie('loginInfo', info)
-  }
+  const setUser = useCallback((arg) => {
```
Comment (cgokey) [inline static/src/js/providers/AuthContextProvider/AuthContextProvider.jsx:35]: I remember we had this code, looks like somewhere along the line it got reverted.

File: static/src/js/providers/AuthContextProvider/AuthContextProvider.jsx [hunk 4]
```diff
@@ -32,12 +32,16 @@ const AuthContextProvider = ({ children }) => {
...
+    if (typeof arg === 'function') {
+      const result = arg(loginInfo)
+      setCookie('loginInfo', result)
+    } else {
+      setCookie('loginInfo', arg)
+    }
+  }, [cookies])
```
Comment (cgokey) [inline static/src/js/providers/AuthContextProvider/AuthContextProvider.jsx:42]: why this dependency?

File: static/src/js/App.jsx (hunk 1 above)
Comment (tle1989) [reply static/src/js/App.jsx:69]: From PR's review: this fix is in another ticket later

File: static/src/js/components/AuthCallbackContainer/AuthCallbackContainer.jsx [hunk 5]
```diff
@@ -12,16 +12,17 @@ import useAppContext from '../../hooks/useAppContext'
...
   const [searchParams] = useSearchParams()
   const { updateLoginInfo, user } = useAppContext()
+
   const path = searchParams.get('target')
   const auid = searchParams.get('auid')
 
   const navigate = useNavigate()
 
   useEffect(() => {
+    updateLoginInfo(auid)
```
[the remark this answers is not in the document]
Comment (tle1989) [reply static/src/js/components/AuthCallbackContainer/AuthCallbackContainer.jsx:22]: This was a fix by Trevor last week due the issue of constant API calls

File: static/src/js/providers/AuthContextProvider/AuthContextProvider.jsx (hunk 3 above)
Comment (tle1989) [reply static/src/js/providers/AuthContextProvider/AuthContextProvider.jsx:35]: This one is also another fix from Trevor last week with constant API call issue

File: static/src/js/App.jsx (hunk 2 above)
Comment (tle1989) [reply static/src/js/App.jsx]: fixed

File: static/src/js/App.jsx (hunk 1 above)
Comment (dmistry1) [reply static/src/js/App.jsx:69]: This change still needs to happen. `typical` will not work in SIT.

**Review by cgokey — APPROVED** [review]

**Pull request merged**